### PR TITLE
Login and registration logic and endpoints

### PR DESCRIPTION
**Pull request by RafMat159**
**DONE:**
_CLIENT & EDITORIAL:_
Added:
* Service layer and controller layer for registration and login
* DTO classes for login and registration
* GlobalExceptionHandler for displaying input warnings during login and registration
* Constants class

_CLIENT_ 
Added:
* Ability to log in to the service using a google account (OAuth2)
* ExternalAuthenticationException to display the information that the user should log in via google

_OTHER:_
* Little changes to SQL files

**TODO:**
Tests to the created methods

### CODE REVIEW

**Review by SzymekN — APPROVED** [review]
looks good

File: client/src/main/java/com/client/controller/LoginController.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        accessToken = jsonObject.getString("access_token");
+        return accessToken;
+    }
+
+    private ResponseEntity<String> getUserInfoResponse(RestTemplate restTemplate, String accessToken) {
+        HttpHeaders accessHeaders = new HttpHeaders();
+        accessHeaders.setBearerAuth(accessToken);
+        HttpEntity<String> accessEntity = new HttpEntity<>("", accessHeaders);
+        return restTemplate.exchange(GOOGLE_USERINFO_ENDPOINT, HttpMethod.GET, accessEntity, String.class);
+    }
```
Comment (xDekann) [inline client/src/main/java/com/client/controller/LoginController.java:155]: Maybe we can try to move it to service layer as well? Just for the sake of keeping only endpoints in controllers. We might consider creating another service layer related class.

File: client/src/main/java/com/client/model/dto/UserRegistrationDto.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    private String surname;
+
+    @Email(regexp = ".+[@].+[\\.].+", message = "Invalid address!")
+    @Size(max = 45, message = "Email must contain less than 46 characters!")
+    @NotBlank(message = "Email must not be blank!")
+    private String email;
+
+    private String supplier = APP_SUPPLIER;
+
+    private String authorityName = ROLE_USER;
```
Comment (xDekann) [inline client/src/main/java/com/client/model/dto/UserRegistrationDto.java:48]: I am not sure if there is a point of creating variable which is pointing to constant static references. Just use static reference in methods instead (so APP_SUPPLIER for example alone).

File: client/src/main/java/com/client/util/Constants.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package com.client.util;
+
+public final class Constants {
+    public static final String ROLE_USER = "USER";
+    public static final String GOOGLE_SUPPLIER = "GOOGLE";
+    public static final String APP_SUPPLIER = "APP";
+    public static final String EDITORIAL_REGISTRATION_URL = "http://gateway:8080/editorial/registration/from-client";
+    //    public static final String EDITORIAL_REGISTRATION_URL = "http://localhost:8080/editorial/registration/from-client";
```
Comment (xDekann) [inline client/src/main/java/com/client/util/Constants.java:8]: I would delete this comment if it's only about gateway/localhost swap.

File: editorial/src/main/java/com/editorial/util/Constants.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.editorial.util;
+
+public final class Constants {
+    public static final String APP_SUPPLIER = "APP";
+    public static final String CLIENT_REGISTRATION_URL = "http://gateway:8080/client/registration/from-editorial";
+//    public static final String CLIENT_REGISTRATION_URL = "http://localhost:8080/client/registration/from-editorial";
```
Comment (xDekann) [inline editorial/src/main/java/com/editorial/util/Constants.java:6]: Not needed comment again

File: editorial/src/main/java/com/editorial/controller/RegisterController.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @PostMapping("/registration")
+    public ResponseEntity<String> createUserAccount(@Valid @RequestBody UserRegistrationDto userRegistrationDto){
+        if(registerService.checkIfUserExistsByEmail(userRegistrationDto.getEmail()))
+            return ResponseEntity.status(HttpStatus.CONFLICT).body("Provided e-mail is already taken!");
+
+        RestTemplate restTemplate = new RestTemplate();
+        registerService.registerUser(userRegistrationDto);
+        restTemplate.postForEntity(CLIENT_REGISTRATION_URL,userRegistrationDto, String.class);
```
Comment (xDekann) [inline editorial/src/main/java/com/editorial/controller/RegisterController.java:35]: "space" after CLIENT_REGISTRATION_URL, ... just a small code cleaning part

File: editorial/src/main/java/com/editorial/util/GlobalExceptionHandler.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+@ControllerAdvice
+public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(MethodArgumentNotValidException ex, HttpHeaders headers,
+                                                                  HttpStatusCode status, WebRequest request) {
+
```
Comment (xDekann) [inline editorial/src/main/java/com/editorial/util/GlobalExceptionHandler.java:22]: New line is kinda not needed here (code cleaning)

File: client/src/main/java/com/client/util/GlobalExceptionHandler.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+@ControllerAdvice
+public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(MethodArgumentNotValidException ex, HttpHeaders headers,
+                                                                  HttpStatusCode status, WebRequest request) {
+
```
Comment (xDekann) [inline client/src/main/java/com/client/util/GlobalExceptionHandler.java:22]: Not needed newline (code cleaning)

File: client/src/main/java/com/client/controller/RegisterController.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    @PostMapping("/registration")
+    public ResponseEntity<String> createUserAccount(@Valid @RequestBody UserRegistrationDto userRegistrationDto) {
+        if (loginService.checkIfUserExistsByEmail(userRegistrationDto.getEmail()))
+            return ResponseEntity.status(HttpStatus.CONFLICT).body("Provided e-mail is already taken!");
+
+        RestTemplate restTemplate = new RestTemplate();
+        registerService.registerUser(userRegistrationDto);
+        restTemplate.postForEntity(EDITORIAL_REGISTRATION_URL,userRegistrationDto, String.class);
```
Comment (xDekann) [inline client/src/main/java/com/client/controller/RegisterController.java:38]: "Space" after EDITORIAL_REGISTRATION_URL, (code cleaning)

File: client/src/main/java/com/client/controller/LoginController.java [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+                + "?response_type=code"
+                + "&client_id=" + clientId
+                + "&redirect_uri=" + URLEncoder.encode(redirectUri, StandardCharsets.UTF_8)
+                + "&scope=openid%20email%20profile";
+        return ResponseEntity.status(HttpStatus.FOUND).location(URI.create(googleAuthUrl)).build();
+    }
+
+    @GetMapping("/login/oauth2/code/google")
+    public ResponseEntity<String> getAuthorizationGoogle(@RequestParam("code") String code, HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse) throws JSONException {
+
```
Comment (xDekann) [inline client/src/main/java/com/client/controller/LoginController.java:100]: Not needed space (code cleaning)

**Pull request merged**